tests/test_csv.py

```diff
@@ -0,0 +1,166 @@
+import re
+from copy import copy
+from unittest import mock
+
+import pytest
+from crate.sphinx import csv
+from crate.sphinx.csv import CSVFilterDirective
+from docutils.statemachine import StringList
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        (1, 1),
+        ("1", 1),
+        (-1, ValueError),
+        ("-1", ValueError)
+    ]
+)
+def test_non_negative_int(input, expected):
+    if expected == ValueError:
+        with pytest.raises(ValueError):
+            csv.non_negative_int(input)
+    else:
+        assert csv.non_negative_int(input) == expected
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    [
+        ("1,2,3,4", [1, 2, 3, 4]),
+        ("a,2,3,4", ValueError),
+        ("-1,2,3,4", ValueError),
+        ("0,-2,3,4", ValueError),
+    ]
+)
+def test_non_negative_int_list(input, expected):
+    if expected == ValueError:
+        with pytest.raises(ValueError):
+            csv.non_negative_int_list(input)
+    else:
+        assert csv.non_negative_int_list(input) == expected
+
+
+directive_params = ['Company,Contact,Country,Attend?'], None, 'source'
+
+directive = CSVFilterDirective(
+    None, None, None, None, None, None, None, None, None)
+tmp_directive_options = {
+    'header': 'Company,Contact,Country,Attend?',
+    'delim': '\t',
+    'file': 'example.csv',
+    'included_cols': [0, 1, 2]
+}
+
+
+def test_csv_filter_exclude_directive():
+    directive_options = copy(tmp_directive_options)
+    directive_options['exclude'] = {3: '(?i)Y\\w*'}
```

Comment:
Let's use raw strings for regex strings. That means you can get rid of the escaping. Same for the other cases below.
```suggestion
    directive_options['exclude'] = {3: r'(?i)Y\w*'}
```